### PR TITLE
Reduce e2e complexity

### PR DESCRIPTION
**Pull request by sjenning**
This simplifies the e2e code structure and allow us to make use of the `-test.run` filter from our CI jobs

### CODE REVIEW

File: test/e2e/autorepair.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
+package e2e
```
Comment (alvaroaleman) [inline test/e2e/autorepair.go:1]: If we want to be able to have a unit test job that does `go test ./...` we need to keep the build tags

Comment (ironcladlou) [reply test/e2e/autorepair.go:1]: Agree, this is why I added the tag in the first place

File: test/e2e/create_cluster.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+)
+
+type TestCreateClusterOptions struct {
+	AWSCredentialsFile string
+	AWSRegion          string
+	PullSecretFile     string
+	ReleaseImage       string
+	ArtifactDir        string
+	BaseDomain         string
+	CPOImage           string
```
Comment (ironcladlou) [inline test/e2e/create_cluster.go:23]: I don't see this being assigned anywhere...

File: test/e2e/autoscaling.go [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+)
+
+type TestAutoscalingOptions struct {
+	AWSCredentialsFile string
+	AWSRegion          string
+	PullSecretFile     string
+	ReleaseImage       string
+	ArtifactDir        string
+	BaseDomain         string
+	CPOImage           string
```
[the remark this answers is not in the document]
Comment (ironcladlou) [reply test/e2e/autoscaling.go:27]: Spot check revealed a couple of these and found a couple of the same issue, might want to audit them all

File: test/e2e/autorepair.go (hunk 1 above)
Comment (sjenning) [reply test/e2e/autorepair.go]: fixed

**Comment by ironcladlou** [comment]
Thanks, this is a lot cleaner!

/lgtm
/hold

**Pull request merged**